app/components/candidate_interface/degree_required_component.rb

```diff
@@ -0,0 +1,73 @@
+class CandidateInterface::DegreeRequiredComponent < ViewComponent::Base
+  CANDIDATE_GRADES = {
+    'Upper second-class honours (2:1)' => 4,
+    'Lower second-class honours (2:2)' => 3,
+    'Third-class honours' => 2,
+    'Pass' => 1,
+  }.freeze
+
+  COURSE_REQUIRED_GRADES = {
+    'two_one' => 4,
+    'two_two' => 3,
+    'third_class' => 2,
+  }.freeze
+
+  COURSE_REQUIRED_GRADE_TEXT = {
+    'two_one' => '2:1 degree or higher (or equivalent)',
+    'two_two' => '2:2 degree or higher (or equivalent)',
+    'third_class' => 'Third-class degree or higher (or equivalent)',
+    'not_required' => 'Any degree grade',
+  }.freeze
+
+  DEGREE_GRADE_INSUFFICIENT_TEXT = {
+    'Lower second-class honours (2:2)' => 'a 2:2 degree.',
+    'Third-class honours' => 'a Third class degree.',
+    'Pass' => 'an Ordinary degree (pass).',
+  }.freeze
+
+  attr_accessor :application_choice, :course_degree_requirements, :uk_bachelor_degrees
+
+  def initialize(application_choice)
+    @application_choice = application_choice
+    @course_degree_requirements = application_choice.current_course.degree_grade
+    @uk_bachelor_degrees = find_uk_bachelor_degrees(application_choice)
+  end
+
+  def candidate_degree_grade_below_required_grade?
+    return false if @uk_bachelor_degrees.empty? || degrees_with_other_grades_only? || bachelors_and_masters_degree?
+
+    COURSE_REQUIRED_GRADES[@course_degree_requirements] > CANDIDATE_GRADES[highest_degree_grade]
+  end
+
+private
+
+  def highest_degree_grade
+    @highest_degree_grade ||= return_highest_degree_grade
+  end
+
+  def return_highest_degree_grade
+    grades = uk_bachelor_degrees.map(&:grade)
+    CANDIDATE_GRADES.each_key { |grade| return grade if grades.include?(grade) }
```

Comment:
I was a bit thrown by having a `return` statement within an iterator – took me a while to work out how this worked. Is it a bit unexpected that it relies on the order of the keys within `CANDIDATE_GRADES` rather than the number value?

Would an explicit sort_by on the integer values be clearer?

```suggestion    
    grades.sort_by {|grade|  CANDIDATE_GRADES[grade] }.last
```

Reply:
Just had a play around with this and turns iut the `return` in the expression was actually causing it to fail!

I agree I think your way is slightly clearer. Just plugged it in but the linter complained and asked that it instead we use:

`grades.max_by { |grade| CANDIDATE_GRADES[grade] }`

It accomplishes the same as the above :)

Reply:
@george-bit nice! Did not even know `max_by` was a method... 😄